File: shoring/flywheel/src/test/java/com/panopset/tests/flywheel/ListAuditTest.java
package com.panopset.tests.flywheel;

import static org.junit.jupiter.api.Assertions.assertEquals;
import java.util.Arrays;
import java.util.List;
import org.junit.jupiter.api.Test;

import com.panopset.flywheel.ListAudit;

public class ListAuditTest {
  
  @Test
  void testListAudit() {
    ListAudit la = new ListAudit();
    la.add("foo", Arrays.asList(new String[]{"a", "b"}));
    la.add("bar", Arrays.asList(new String[]{"b", "c"}));
    la.add("zlist", Arrays.asList(new String[]{"x", "y"}));
    la.add("alist", Arrays.asList(new String[]{"b", "a"}));
    List<String> report = la.getReport();
    assertEquals(6, report.size());
    assertEquals(",alist,bar,foo,zlist", report.get(0));
    assertEquals("a,*,,*,", report.get(1));
    assertEquals("b,*,*,*,", report.get(2));
    assertEquals("c,,*,,", report.get(3));
    assertEquals("x,,,,*", report.get(4));
    assertEquals("y,,,,*", report.get(5));
  }
}

File: code/shoring/flywheel/src/test/kotlin/com/panopset/tests/flywheel/ListAuditTest.kt
package com.panopset.tests.flywheel

import com.panopset.flywheel.ListAudit
import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test
import java.util.*

class ListAuditTest {
    @Test
    fun testListAudit() {
        val la = ListAudit()
        la.add("foo", Arrays.asList(*arrayOf("a", "b")))
        la.add("bar", Arrays.asList(*arrayOf("b", "c")))
        la.add("zlist", Arrays.asList(*arrayOf("x", "y")))
        la.add("alist", Arrays.asList(*arrayOf("b", "a")))
        val report = la.report
        Assertions.assertEquals(6, report.size)
        Assertions.assertEquals(",alist,bar,foo,zlist", report[0])
        Assertions.assertEquals("a,*,,*,", report[1])
        Assertions.assertEquals("b,*,*,*,", report[2])
        Assertions.assertEquals("c,,*,,", report[3])
        Assertions.assertEquals("x,,,,*", report[4])
        Assertions.assertEquals("y,,,,*", report[5])
    }
}
